tests/bgp/test_bgp_update_timer.py

```diff
@@ -35,18 +35,18 @@
 
 
 @contextlib.contextmanager
-def log_bgp_updates(duthost, iface, save_path):
+def log_bgp_updates(duthost, iface, save_path, ns):
     """Capture bgp packets to file."""
     if iface == "any":
         # Scapy doesn't support LINUX_SLL2 (Linux cooked v2), and tcpdump on Bullseye
         # defaults to writing in that format when listening on any interface. Therefore,
         # have it use LINUX_SLL (Linux cooked) instead.
         start_pcap = "tcpdump -y LINUX_SLL -i %s -w %s port 179" % (iface, save_path)
     else:
-        start_pcap = "tcpdump -i %s -w %s port 179" % (iface, save_path)
-    stop_pcap = "pkill -f '%s'" % start_pcap
-    start_pcap = "nohup %s &" % start_pcap
-    duthost.shell(start_pcap)
+        start_pcap = "tcpdump -i %s -w %s port 179" % (iface, save_path)  
+    # for multi-asic dut, add 'ip netns exec asicx' to the beggining of tcpdump cmd 
+    stop_pcap = "sudo pkill -f '%s %s'" % (duthost.asic_instance_from_namespace(ns).ns_arg, start_pcap)
```

Comment:
Will this still work in the single-ASIC case? I'm guessing `duthost.asic_instance_from_namespace(ns).ns_arg` evaluates to an empty string in the single-ASIC case, in which case the command will be `sudo pkill -f ' %s'` (with the space before the `%s`).

Reply:
thanks @saiarcot895, good catch!

Reply:
@saiarcot895 
new change is working on single-asic dut too, attached log in PR description.
thanks for pointing out!